src/components/Modal.jsx

```diff
@@ -0,0 +1,135 @@
+import "flowbite";
+import "./Modal.css";
+import CrossSvg from "./CrossSvg";
+import { useEffect, useState } from "react";
+import RenderIf from "./RenderIf";
+
+function FormElement(props) {
+  const [state, setState] = useState(props.value[props.index]);
+
+  useEffect(() => {
+    setState(props.value[props.index]);
+  }, [props.value[props.index]]);
+
+  return (
+    <div>
+      <label
+        for={props.propKey}
+        class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
+      >
+        {props.propKey}
+      </label>
+      <RenderIf
+        isTrue={props.disabled}
+        fallback={
+          <input
+            type={props.type}
+            name={props.propKey}
+            id={props.propKey}
+            class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white"
+            value={state}
+            onChange={(e) => setState(e.target.value)}
+            required
+          />
+        }
+      >
```

Comment:
We can also write it like this 
        <input
          type={props.type}
          name={props.propKey}
          id={props.propKey}
          class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white"
          value={props.propKey === "id" ? state || props.id : state}
          onChange={(e) => setState(e.target.value)}
          required
          disabled={props.disabled}
        />